frontend/lib/parsing/parsing-queries.cpp

```diff
@@ -1377,6 +1377,22 @@ static std::string hardcodedDeprecationForId(Context* context, ID idMention,
   return deprecationMsg;
 }
 
+static bool parentIgnoresDeprecated(Context* context, ID idTarget) {
+  auto parentId = parsing::idToParentId(context, idTarget);
+  if (auto attrib = parsing::idToAttributeGroup(context, parentId)) {
+    if (attrib->hasPragma(PRAGMA_IGNORE_DEPRECATED_USE)) return true;
+  } else if (parentId.isEmpty()) {
+    return false;
+  }
+
+  auto ast = parsing::idToAst(context, parentId);
+  if (ast->isModule()) {
+    return false;
+  }
+
+  return parentIgnoresDeprecated(context, parentId);
+}
```

Comment:
Can't you use David's `shouldSkipDeprecationWarning` to handle to upwards traversal for you? That function seems to handle other "skip if parent is XYZ" stuff.

Reply:
Absolutely I should!